agentops/time_travel.py

```diff
@@ -0,0 +1,137 @@
+import json
+import yaml
+from .http_client import HttpClient
+import os
+from .helpers import singleton
+from dotenv import load_dotenv
+from .log_config import logger
+from os import environ
+
+
+@singleton
+class TimeTravel:
+    def __init__(self):
+        self._completion_overrides_map = {}
+        self._prompt_override_map = {}
+
+        script_dir = os.path.dirname(os.path.abspath(__file__))
+        parent_dir = os.path.dirname(script_dir)
+        cache_path = os.path.join(parent_dir, "agentops_time_travel.json")
+
+        try:
+            with open(cache_path, "r") as file:
+                time_travel_cache_json = json.load(file)
+                self._completion_overrides_map = time_travel_cache_json.get(
+                    "completion_overrides"
+                )
+                self._prompt_override_map = time_travel_cache_json.get(
+                    "prompt_override"
+                )
+        except FileNotFoundError:
+            return
+
+
+def fetch_time_travel_id(ttd_id):
+    try:
+        endpoint = environ.get("AGENTOPS_API_ENDPOINT", "https://api.agentops.ai")
+        payload = json.dumps({"ttd_id": ttd_id}).encode("utf-8")
+        ttd_res = HttpClient.post(f"{endpoint}/v2/get_ttd", payload)
```

Comment:
this post needs to be wrapped in a try/catch and handle `except ApiServerException:`. this is a recent change

Reply:
Done